File: rust/src/models/subscription_fee.rs
// this file is @generated
use serde::{Deserialize, Serialize};

use super::{
    capacity_fee::CapacityFee, one_time_fee::OneTimeFee, rate_fee::RateFee,
    recurring_fee::RecurringFee, slot_fee::SlotFee, usage_fee::UsageFee,
};

#[derive(Clone, Debug, PartialEq, Deserialize, Serialize)]
#[serde(tag = "discriminator")]
pub enum SubscriptionFee {
    #[serde(rename = "RATE")]
    Rate(RateFee),
    #[serde(rename = "ONE_TIME")]
    OneTime(OneTimeFee),
    #[serde(rename = "RECURRING")]
    Recurring(RecurringFee),
    #[serde(rename = "CAPACITY")]
    Capacity(CapacityFee),
    #[serde(rename = "SLOT")]
    Slot(SlotFee),
    #[serde(rename = "USAGE")]
    Usage(UsageFee),
}

impl Default for SubscriptionFee {
    fn default() -> Self {
        Self::Rate(RateFee::default())
    }
}

File: python/meteroid/models/subscription_fee.py
# this file is @generated
import dataclasses
import typing as t

from ..serialization import BaseModel, TaggedUnionModel
from .capacity_fee import CapacityFee
from .one_time_fee import OneTimeFee
from .rate_fee import RateFee
from .recurring_fee import RecurringFee
from .slot_fee import SlotFee
from .usage_fee import UsageFee


@dataclasses.dataclass
class SubscriptionFee(TaggedUnionModel):
    _DISCRIMINATOR: t.ClassVar[str] = "type"
    _DISCRIMINATOR_ATTR: t.ClassVar[str] = "type"
    _CONTENT_ATTR: t.ClassVar[str] = "content"
    _CONTENT_KEY: t.ClassVar[t.Optional[str]] = None
    _VARIANTS: t.ClassVar[t.Mapping[str, t.Optional[t.Type[BaseModel]]]] = {
        "RATE": RateFee,
        "ONE_TIME": OneTimeFee,
        "RECURRING": RecurringFee,
        "CAPACITY": CapacityFee,
        "SLOT": SlotFee,
        "USAGE": UsageFee,
    }

    type: t.Literal[
        "RATE",
        "ONE_TIME",
        "RECURRING",
        "CAPACITY",
        "SLOT",
        "USAGE",
    ]
    content: t.Union[
        RateFee,
        OneTimeFee,
        RecurringFee,
        CapacityFee,
        SlotFee,
        UsageFee,
    ]
